simple/lint.go

```diff
@@ -68,6 +68,43 @@ func (c *Checker) Funcs() map[string]lint.Func {
 		"S1028": c.LintErrorsNewSprintf,
 		"S1029": c.LintRangeStringRunes,
 		"S1030": c.LintBytesBufferConversions,
+		"S1031": c.LintNilCheckAroundRange,
+	}
+}
+
+func (c *Checker) LintNilCheckAroundRange(j *lint.Job) {
```

Comment:
This check is missing one additional condition: We can only suggest the simplification if ranging over a slice.

In the case of channels, ranging over a nil channel would block forever, which is why the check is needed.

In the case of maps, the loop body may try to insert values in the map. Trying to do that on a nil map panics.